benchbuild/source/base.py

```diff
@@ -0,0 +1,126 @@
+"""
+Provide a base interface for downloadable sources.
+"""
+import abc
+import itertools
+from typing import Iterable, List, Mapping, Union
+
+import attr
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unable to import 'attr'](https://app.codacy.com/manual/polyjit/benchbuild_2/pullRequest?prid=5694346)